v1/plugins/logs/eventBuffer_test.go

```diff
@@ -0,0 +1,252 @@
+package logs
+
+import (
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/open-policy-agent/opa/v1/ast"
+	"github.com/open-policy-agent/opa/v1/keys"
+	"github.com/open-policy-agent/opa/v1/logging"
+	"github.com/open-policy-agent/opa/v1/metrics"
+	"github.com/open-policy-agent/opa/v1/plugins/rest"
+	"github.com/open-policy-agent/opa/v1/topdown/builtins"
+)
+
+func TestEventBuffer_Push(t *testing.T) {
+	t.Parallel()
+
+	expectedIds := make(map[string]struct{})
+	var expectedDropped uint64
+	limit := int64(2)
+	b := newEventBuffer(limit, rest.Client{}, "", 0)
+	p := &Plugin{
+		metrics: metrics.New(),
+	}
+
+	id := "id1"
+	expectedIds[id] = struct{}{}
+	b.Push(p, newTestEvent(t, id, false))
+	checkBufferState(t, limit, p, b, expectedDropped, expectedIds)
+
+	id = "id2"
+	expectedIds[id] = struct{}{}
+	b.Push(p, newTestEvent(t, id, false))
+	checkBufferState(t, limit, p, b, expectedDropped, expectedIds)
+
+	id = "id3"
+	expectedIds[id] = struct{}{}
+	b.Push(p, newTestEvent(t, id, false))
+	// Three events were pushed, but limit is 2 so the oldest even should have been dropped
+	delete(expectedIds, "id1")
+	expectedDropped++
+	checkBufferState(t, limit, p, b, expectedDropped, expectedIds)
+
+	if int64(len(b.buffer)) != limit {
+		t.Fatalf("buffer size mismatch, expected %d, got %d", limit, len(b.buffer))
+	}
+
+	// Increase the limit, forcing the buffer to change
+	limit = int64(3)
+	b.Reconfigure(p, limit, rest.Client{}, "", 0)
+	checkBufferState(t, limit, p, b, expectedDropped, expectedIds)
+
+	id = "id4"
+	expectedIds[id] = struct{}{}
+	b.Push(p, newTestEvent(t, id, false))
+	checkBufferState(t, limit, p, b, expectedDropped, expectedIds)
+
+	id = "id5"
+	expectedIds[id] = struct{}{}
+	b.Push(p, newTestEvent(t, id, true))
+	// Four events were pushed, but limit is 3 so the oldest even should have been dropped
+	expectedDropped++
+	delete(expectedIds, "id2")
+	checkBufferState(t, limit, p, b, expectedDropped, expectedIds)
+
+	limit = int64(1)
+	b.Reconfigure(p, limit, rest.Client{}, "", 0)
+	// Limit reconfigured from 3->1, dropping 2 more events.
+	expectedDropped = 4
+	delete(expectedIds, "id3")
+	delete(expectedIds, "id4")
+	checkBufferState(t, limit, p, b, expectedDropped, expectedIds)
+
+	// Nothing changed
+	b.Reconfigure(p, limit, rest.Client{}, "", 0)
+	checkBufferState(t, limit, p, b, expectedDropped, expectedIds)
+}
+
+func checkBufferState(t *testing.T, limit int64, p *Plugin, b *eventBuffer, expectedDropped uint64, expectedIds map[string]struct{}) {
```

Comment:
Adding `t.Helper()` to the top of this helper function will make the testing framework report the call-site as the error line instead of the `t.Fatalf()` calls inside this function.